discord/guild.go

```diff
@@ -1,57 +1,124 @@
+	// ID is the guild id.
+	ID Snowflake `json:"id,string"`
+	// Name is the guild name (2-100 characters, excluding trailing and leading
+	// whitespace).
+	Name string `json:"name"`
+	// Icon is the icon hash.&nullableUint64{}
+	Icon Hash `json:"icon"`
+	// Splash is the splash hash.
+	Splash Hash `json:"splash,omitempty"`
+	// DiscoverySplash is the discovery splash hash.
+	//
+	// Only present for guilds with the "DISCOVERABLE" feature.
+	DiscoverySplash Hash `json:"discovery_splash,omitempty"`
 
-	Owner   bool      `json:"owner,omitempty"` // self is owner
+	// Owner is true if the user is the owner of the guild.
+	Owner bool `json:"owner,omitempty"`
+	// OwnerID is the id of owner.
 	OwnerID Snowflake `json:"owner_id,string"`
 
+	// Permissions are the total permissions for the user in the guild
+	// (excludes overrides).
 	Permissions Permissions `json:"permissions,omitempty"`
 
+	// VoiceRegion is the voice region id for the guild.
 	VoiceRegion string `json:"region"`
 
+	// AFKChannelID is the id of the afk channel.
 	AFKChannelID Snowflake `json:"afk_channel_id,string,omitempty"`
-	AFKTimeout   Seconds   `json:"afk_timeout"`
-
-	Embeddable     bool      `json:"embed_enabled,omitempty"`
+	// AFKTimeout is the afk timeout in seconds.
+	AFKTimeout Seconds `json:"afk_timeout"`
+
+	// Embeddable is true if the server widget is enabled.
+	//
+	// Deprecated: replaced with WidgetEnabled
+	Embeddable bool `json:"embed_enabled,omitempty"`
+	// EmbedChannelID is the channel id that the widget will generate an invite
+	// to, or null if set to no invite .
+	//
+	// Deprecated: replaced with WidgetChannelID
 	EmbedChannelID Snowflake `json:"embed_channel_id,string,omitempty"`
 
-	Verification   Verification   `json:"verification_level"`
-	Notification   Notification   `json:"default_message_notifications"`
+	// Verification is the verification level required for the guild.
+	Verification Verification `json:"verification_level"`
+	// Notification is the default message notifications level.
+	Notification Notification `json:"default_message_notifications"`
+	// ExplicitFilter is the explicit content filter level.
 	ExplicitFilter ExplicitFilter `json:"explicit_content_filter"`
 
-	Roles    []Role         `json:"roles"`
-	Emojis   []Emoji        `json:"emojis"`
+	// Roles are the roles in the guild.
+	Roles []Role `json:"roles"`
+	// Emojis are the custom guild emojis.
+	Emojis []Emoji `json:"emojis"`
+	// Features are the enabled guild features.
 	Features []GuildFeature `json:"guild_features"`
 
+	// MFA is the required MFA level for the guild.
 	MFA MFALevel `json:"mfa"`
 
+	// AppID is the application id of the guild creator if it is bot-created.
+	//
+	// This field is nullable.
 	AppID Snowflake `json:"application_id,string,omitempty"`
 
-	Widget bool `json:"widget_enabled,omitempty"`
-
+	// WidgetEnabled is true if the server widget is enabled.
+	WidgetEnabled bool `json:"widget_enabled,omitempty"`
```

Comment:
Can this be changed back to `Widget`? I don't really see the justification for breaking the existing API.